pkg/tasks/c1api/bulk_get_tickets.go

```diff
@@ -0,0 +1,66 @@
+package c1api
+
+import (
+	"context"
+	"errors"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	v1 "github.com/conductorone/baton-sdk/pb/c1/connectorapi/baton/v1"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/types"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+)
+
+type bulkGetTicketsTaskHelpers interface {
+	ConnectorClient() types.ConnectorClient
+	FinishTask(ctx context.Context, resp proto.Message, annos annotations.Annotations, err error) error
+}
+
+type bulkGetTicketTaskHandler struct {
+	task    *v1.Task
+	helpers bulkGetTicketsTaskHelpers
+}
+
+func (c *bulkGetTicketTaskHandler) HandleTask(ctx context.Context) error {
+	l := ctxzap.Extract(ctx)
+
+	cc := c.helpers.ConnectorClient()
+
+	t := c.task.GetBulkGetTickets()
+	if t == nil || len(t.GetTicketRequests()) == 0 {
+		l.Error("bulk get ticket task was nil or missing tickets", zap.Any("bulk_get_tickets_task", t))
+		return c.helpers.FinishTask(ctx, nil, nil, errors.Join(errors.New("malformed bulk get tickets task"), ErrTaskNonRetryable))
+	}
+
+	ticketRequests := make([]*v2.TicketsServiceGetTicketRequest, 0)
+	for _, getTicketTask := range t.GetTicketRequests() {
+		ticketRequests = append(ticketRequests, &v2.TicketsServiceGetTicketRequest{
+			Id:          getTicketTask.GetTicketId(),
+			Annotations: getTicketTask.GetAnnotations(),
+		})
+	}
+
+	resp, err := cc.BulkGetTickets(ctx, &v2.TicketsServiceBulkGetTicketRequest{
+		TicketRequests: ticketRequests,
+	})
+	if err != nil {
+		return c.helpers.FinishTask(ctx, nil, nil, err)
+	}
+
+	if len(resp.GetTickets()) == 0 {
```

Comment:
should this be a comparison to the length of `ticketRequests` ?

Reply:
yea that makes more sense